internal/controller/etcdcluster_controller.go

```diff
@@ -84,41 +84,54 @@ func (r *EtcdClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 	}()
 
-	if err := r.ensureClusterObjects(ctx, instance); err != nil {
+	// check sts condition
+	isClusterReady := false
+	sts := &appsv1.StatefulSet{}
+	err = r.Get(ctx, client.ObjectKey{
+		Namespace: instance.Namespace,
+		Name:      instance.Name,
+	}, sts)
+	if err == nil {
+		isClusterReady = sts.Status.ReadyReplicas == *sts.Spec.Replicas
+	}
+
+	if err := r.ensureClusterObjects(ctx, instance, isClusterReady); err != nil {
 		return ctrl.Result{}, fmt.Errorf("cannot create Cluster auxiliary objects: %w", err)
 	}
 
-	if initIdx := slices.IndexFunc(instance.Status.Conditions, func(condition metav1.Condition) bool {
-		return condition.Type == etcdaenixiov1alpha1.EtcdConditionInitialized
-	}); initIdx != -1 {
-		instance.Status.Conditions[initIdx].Status = "True"
-		instance.Status.Conditions[initIdx].LastTransitionTime = metav1.Now()
-		instance.Status.Conditions[initIdx].Reason = "InitializationComplete"
-		instance.Status.Conditions[initIdx].Message = "Cluster initialization is complete"
-	} else {
-		instance.Status.Conditions = append(instance.Status.Conditions, metav1.Condition{
-			Type:               etcdaenixiov1alpha1.EtcdConditionInitialized,
+	r.updateClusterState(instance, metav1.Condition{
+		Type:               etcdaenixiov1alpha1.EtcdConditionInitialized,
+		Status:             "True",
+		LastTransitionTime: metav1.Now(),
+		Reason:             "InitializationComplete",
```

Comment:
IMHO better us a a const in an api package 
for instance:
/api/v1alpha1
```
type ETCDConditionType string
const (
	ETCDCondTypeInitComplete    ETCDConditionType = "InitializationComplete"
)
```

Reason: string(etcdaenixiov1alpha1.ETCDCondTypeInitComplete)